packages/template-retail-react-app/app/components/product-view/index.test.js

```diff
@@ -222,3 +223,27 @@ test('onVariantSelected callback is called after successfully selected a variant
         expect(onVariantSelected).toHaveBeenCalledTimes(1)
     })
 })
+
+describe('add to cart button loading tests', () => {
+    test('add to cart button is disabled if isBasketLoading is true', async () => {
+        renderWithProviders(
+            <MockComponent
+                product={mockProductDetail}
+                addToCart={() => {}}
+                isBasketLoading={true}
+            />
+        )
+        expect(screen.getByRole('button', {name: /add to cart/i})).toBeDisabled()
+    })
+
+    test('add to cart button is enabled if isBasketLoading is false', async () => {
+        renderWithProviders(
+            <MockComponent
+                product={mockProductDetail}
+                addToCart={() => {}}
+                isBasketLoading={false}
+            />
+        )
+        expect(screen.getByRole('button', {name: /add to cart/i})).not.toBeDisabled()
```

Comment:
```suggestion
        expect(screen.getByRole('button', {name: /add to cart/i})).toBeEnabled()
```
🤷

Reply:
Thanks for these suggestions! Changes have been applied